File: frame/map-frame-project/MapFrameCoreModule/src/main/java/com/grandtech/mapframe/core/select/SelectSetting.java
package com.grandtech.mapframe.core.select;

import com.grandtech.mapframe.core.rules.Rules;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.LinkedHashMap;
import java.util.List;

/**
 * Created by zy on 2019/1/3.
 */

public class SelectSetting  implements Rules {

    List<String> layerIds = new ArrayList<>();

    private int maxSelectCount = 20;

    boolean isMulti = true;
    /**
     * 是否优先选择
     */
    boolean isPriority = false;

    /**
     * 是否渲染结果集
     */
    boolean isRender = true;

    private LinkedHashMap layerIdSelectMap;



    public SelectSetting(List<String> layerIds) {
        if (layerIds == null) {
            return;
        }
        this.layerIds.addAll(layerIds);
    }

    public SelectSetting(String... layerIds) {
        if (layerIds == null) {
            return;
        }
        this.layerIds.addAll(Arrays.asList(layerIds));
    }



    public SelectSetting(boolean isPriority, List<String> layerIds) {
        if (layerIds == null) {
            return;
        }
        this.isPriority = isPriority;
        this.layerIds.addAll(layerIds);
    }

    public SelectSetting(boolean _isMulti, boolean isPriority, String... layerIds) {
        if (layerIds == null) {
            return;
        }
        this.isMulti = _isMulti;
        this.isPriority = isPriority;
        this.layerIds.addAll(Arrays.asList(layerIds));
    }



    public SelectSetting(boolean isMulti, boolean isRender, boolean isPriority, List<String> layerIds) {
        if (layerIds == null) {
            return;
        }
        this.isMulti = isMulti;
        this.isRender = isRender;
        this.isPriority = isPriority;
        this.layerIds.addAll(layerIds);
    }

    public SelectSetting(boolean isMulti, boolean isRender, boolean isPriority, String... layerIds) {
        if (layerIds == null) {
            return;
        }
        this.isMulti = isMulti;
        this.isRender = isRender;
        this.isPriority = isPriority;
        this.layerIds.addAll(Arrays.asList(layerIds));
    }


    public SelectSetting(boolean isRender, boolean isPriority, List<String> layerIds) {
        if (layerIds == null) {
            return;
        }
        this.isRender = isRender;
        this.isPriority = isPriority;
        this.layerIds.addAll(layerIds);
    }

    public LinkedHashMap getLayerIdSelectId() {
        return layerIdSelectMap;
    }

    public void setLayerIdSelect(LinkedHashMap layerIdSelectMap) {
        this.layerIdSelectMap = layerIdSelectMap;
    }

    public List<String> getLayersAsList() {
        return layerIds;
    }

    public boolean isPriority() {
        return isPriority;
    }

    public boolean isRender() {
        return isRender;
    }

    public boolean isMulti() {
        return isMulti;
    }

    public int getMaxSelectCount(){return  maxSelectCount;}
}

File: frame/map-frame-project/MapFrameCoreModuleV10/src/main/java/com/gradtech/mapframev10/core/select/SelectSetting.kt
package com.gradtech.mapframev10.core.select

import com.gradtech.mapframev10.core.rules.Rules
import org.jetbrains.annotations.NotNull
import java.util.*

/**
 * @ClassName SelectSetting
 * @Description TODO
 * @Author: fs
 * @Date: 2022/9/8 15:22
 * @Version 2.0
 */
class SelectSetting:Rules {

    var layerIds: MutableList<String> = mutableListOf()

    private val maxSelectCount = 20

    private var isMulti = true

    /**
     * 是否优先选择
     */
    private var isPriority = false

    /**
     * 是否渲染结果集
     */
    private var isRender = true

    /**
     * 是否支持反选
     */
    private var canCancelSelected = true

     constructor(@NotNull layerIds: List<String>) {
        if (layerIds == null) {
            return
        }
        this.layerIds.addAll(layerIds)
    }

     constructor(@NotNull vararg layerIds: String) {
        if (layerIds == null) {
            return
        }
        for (layerId in layerIds) {
            this.layerIds.add(layerId)
        }

    }


     constructor(@NotNull isPriority: Boolean, @NotNull layerIds: List<String>) {
        if (layerIds == null) {
            return
        }
        this.isPriority = isPriority
        this.layerIds.addAll(layerIds)
    }

     constructor(@NotNull _isMulti: Boolean, @NotNull isPriority: Boolean, @NotNull vararg layerIds: String) {
        if (layerIds == null) {
            return
        }
        isMulti = _isMulti
        this.isPriority = isPriority
        for (layerId in layerIds) {
            this.layerIds.add(layerId)
        }

    }
    @JvmOverloads
    constructor(isMulti: Boolean, isRender: Boolean, isPriority: Boolean, layerIds: MutableList<String>?) {
        if (layerIds == null) {
            return
        }
        this.isMulti = isMulti
        this.isRender = isRender
        this.isPriority = isPriority
        this.layerIds.addAll(layerIds)
    }
    @JvmOverloads
     constructor(isMulti: Boolean, isRender: Boolean, isPriority: Boolean, @NotNull vararg layerIds: String) {
        if (layerIds == null) {
            return
        }
        this.isMulti = isMulti
        this.isRender = isRender
        this.isPriority = isPriority
        for (layerId in layerIds) {
            this.layerIds.add(layerId)
        }
    }

    @JvmOverloads
    constructor(isMulti: Boolean, isRender: Boolean, isPriority: Boolean, canCancelSelected :Boolean,layerIds: MutableList<String>?) {
        if (layerIds == null) {
            return
        }
        this.isMulti = isMulti
        this.isRender = isRender
        this.isPriority = isPriority
        this.canCancelSelected = canCancelSelected
        this.layerIds.addAll(layerIds)
    }
    @JvmOverloads
    constructor(isMulti: Boolean, isRender: Boolean, isPriority: Boolean,canCancelSelected :Boolean, @NotNull vararg layerIds: String) {
        if (layerIds == null) {
            return
        }
        this.isMulti = isMulti
        this.isRender = isRender
        this.isPriority = isPriority
        this.canCancelSelected = canCancelSelected
        for (layerId in layerIds) {
            this.layerIds.add(layerId)
        }
    }


     constructor(isRender: Boolean, isPriority: Boolean, layerIds: List<String>?) {
        if (layerIds == null) {
            return
        }
        this.isRender = isRender
        this.isPriority = isPriority
        this.layerIds.addAll(layerIds)
    }



    fun getLayersAsList(): List<String>? {
        return layerIds
    }

    fun isPriority(): Boolean {
        return isPriority
    }

    fun isRender(): Boolean {
        return isRender
    }

    fun isMulti(): Boolean {
        return isMulti
    }

    fun getMaxSelectCount(): Int {
        return maxSelectCount
    }
    fun canCancelSelected(): Boolean {
        return canCancelSelected
    }

}